src/biome/text/trainer.py

```diff
@@ -0,0 +1,124 @@
+import copy
+from dataclasses import asdict
+from typing import Optional
+from typing import Union
+
+import pytorch_lightning as pl
+from allennlp.common import Params
+from allennlp.data import PyTorchDataLoader
+from allennlp.data.samplers import BucketBatchSampler
+from allennlp.training.optimizers import Optimizer
+from torch.utils.data import IterableDataset
+
+from biome.text.configuration import LTrainerConfiguration
+from biome.text.configuration import VocabularyConfiguration
+from biome.text.dataset import Dataset
+from biome.text.dataset import InstancesDataset
+from biome.text.pipeline import Pipeline
+from biome.text.training_results import TrainingResults
+
+
+class Trainer:
+    def __init__(
+        self,
+        config: LTrainerConfiguration,
+    ):
+        self._config = copy.deepcopy(config)
+
+        # remove non pl.Trainer arguments
+        trainer_kwargs = asdict(self._config)
+        for kwarg in ["num_epochs", "optimizer", "data_bucketing", "batch_size"]:
+            del trainer_kwargs[kwarg]
+
+        self.trainer = pl.Trainer(**trainer_kwargs)
+
+    def create_dataloader(
```

Comment:
Is it worth having this method as a helper function in order to be able to use a pytorch trainer directly?

```python
trainer= pl.Trainer(....)

trainer.fit(pipeline.model, helper.create_dataloader(dataset, batch_size=32, data_bucketing=True)
....
```

Reply:
Done